admin/create-theme/theme-utils.php

```diff
@@ -28,4 +28,154 @@ public static function get_file_extension_from_url( $url ) {
 		$extension = pathinfo( $url, PATHINFO_EXTENSION );
 		return $extension;
 	}
+
+	public static function replace_namespace( $content, $new_slug, $new_name ) {
+
+		$old_slug            = wp_get_theme()->get( 'TextDomain' );
+		$new_slug_underscore = str_replace( '-', '_', $new_slug );
+		$old_slug_underscore = str_replace( '-', '_', $old_slug );
+		$old_name            = wp_get_theme()->get( 'Name' );
```

Comment:
Every call to `wp_get_theme` creates a new instance; I think it might be worth keeping just one and reusing it.

```suggestion
		$theme               = wp_get_theme();    
		$old_slug            = $theme->get( 'TextDomain' );
		$new_slug_underscore = str_replace( '-', '_', $new_slug );
		$old_slug_underscore = str_replace( '-', '_', $old_slug );
		$old_name            = $theme->get( 'Name' );
```

Reply:
Going a step further, declaring a static `$theme` on the class could also save some of these calls. 

```php
private static $theme = wp_get_theme();
```

and then use it like

```php
self::$theme;
```

Reply:
Thank you; good call.

That method was used throughout the app prior to this change and it would be good to clean that up in all the places.  I hesitate to do so outside of the bits changed here so instead I've [opened a new issue](https://github.com/WordPress/create-block-theme/issues/379) regarding cleaning up those calls throughout that perhaps we can address in the future.